docs/id-repository-credential-request-generator-service-developer-guide.md

```diff
@@ -0,0 +1,120 @@
+4. pgAdmin
+5. Postman
+6. Git
+7. Notepad++ (optional)
+8. lombok.jar (file)
+9. settings.xml (document)
+
+
+Follow the steps below to set up ID Repository Services on your local system:
+
+1. Download `lombok.jar` and `settings.xml` from [here](https://github.com/mosip/documentation/tree/1.2.0/docs/_files/id-repsoitory-config-files).
+
+2. Unzip Apache Maven and move the unzipped folder in ```C:\Program Files``` and `settings.xml` to "conf" folder `C:\Program Files\apache-maven-3.8.4\conf`.
+
+3. Install Eclipse, open the `lombok.jar` file and wait for some time until it completes the scan for Eclipse IDE and then click `Install/Update`.
+
+<img src="_images/lombok-configuration.png" width="750" height="450">
+
+4. Check the Eclipse installation folder `C:\Users\userName\eclipse\jee-2021-12\eclipse` to see if the `lombok.jar` is added. By doing this, you don't have to add the dependency of `lombok` in your `pom.xml` file separately as it is auto-configured by Eclipse.
+
+5. Configure the JDK (Standard VM) with your Eclipse by traversing through `Preferences → Java → Installed JREs`.
+
+<img src="_images/installed-jre.png" width="750" height="450">
+ 
+## Code setup
+
+For the code setup, clone the repository and follow the guidelines mentioned in the [Code Contributions](https://docs.mosip.io/1.2.0/community/code-contributions).
+
+### Importing and building of a project
+
+1. Open the project folder where `pom.xml` is present.
+1. Open command prompt from the same folder.
+1. Run the command `mvn clean install -Dgpg.skip=true` to build the project and wait for the build to complete successfully.
+1. After building of a project, open Eclipse and select `Import Projects → Maven → Existing Maven Projects → Next → Browse to project directory → Finish`.
+1. After successful importing of project, update the project by right-click on `Project → Maven → Update Project`.
+
+<img src="_images/import-project.png" width="750" height="450">
+
+## Environment setup
+
+1. For the environment setup, you need an external JAR that is available [here](https://oss.sonatype.org/#nexus-search;gav~~kernel-auth-adapter~1.2.0-SNAPSHOT~~) with different versions. (E.g.: You can download `kernel-auth-adapter.jar` and add to project `Libraries → Classpath → Add External JARs → Select Downloaded JAR → Add → Apply and Close`).
+
+<img src="_images/add-external-library.png" width="750" height="450">
+
+2. Clone [mosip-config repository](https://github.com/mosip/mosip-config).
+
+3. Create an empty folder inside the `mosip-config` with `sandbox-local` name and then copy and paste all config files inside `sandbox-local` folder except `.gitignore, README and LICENSE`.
+
+4. As Id Repository is using two properties files, `id-repository-default` and `application-default`, you will have to configure them according to your environment.    The same files are available [here](https://github.com/mosip/documentation/tree/1.2.0/docs/_files/id-repository-config-files) for reference. 
```

Comment:
ID -capitalization

Reply:
It is exactly mentioned as per the name of the "id-repository-default.properties" file.